Add more tests for expected loop_protect_external

grafnu opened the pull request.

anarkiwi commented on tests/integration/mininet_tests.py:
```diff
@@ -6979,6 +6979,17 @@ def verify_tunnel_established(self, src_host, dst_host, other_host, packets=3):
+        for to_host in to_hosts:
+            try:
+                self.verify_broadcast(hosts=(from_host, to_host), broadcast_expected=False)
+            except Exception:
```
What exception are you expecting here?

grafnu commented:
I changed it to AssertionError and added a comment. It's basically catching
the exception of when a broadcast is received (since expected=False), and
counting them.  Other way would be to refactor the test library code to be
able to (optionally) return the broadcast state, rather than force an
assertion.


On Wed, Oct 30, 2019 at 10:27 PM Josh Bailey <notifications@github.com>
wrote:

> *@anarkiwi* commented on this pull request.
> ------------------------------
>
> In tests/integration/mininet_tests.py
> <https://github.com/faucetsdn/faucet/pull/3317#discussion_r340964810>:
>
> > @@ -6979,6 +6979,17 @@ def verify_tunnel_established(self, src_host, dst_host, other_host, packets=3):
>              '%s: ICMP echo request' % other_host.IP(), tcpdump_text
>          ), 'Tunnel was not established')
>
> +    def verify_one_broadcast(self, from_host, to_hosts):
> +        self.assertGreater(len(to_hosts), 1, 'Testing only one ext host is not useful')
> +        received_broadcasts = []
> +        for to_host in to_hosts:
> +            try:
> +                self.verify_broadcast(hosts=(from_host, to_host), broadcast_expected=False)
> +            except Exception:
>
> What exception are you expecting here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/faucetsdn/faucet/pull/3317?email_source=notifications&email_token=AAIEPD5H347ZCCKLPRECIDLQRJUCFA5CNFSM4JHC4RMKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCJ2PLHA#pullrequestreview-309654940>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIEPDZ5GHGD745X3BCBLDLQRJUCFANCNFSM4JHC4RMA>
> .
>

anarkiwi commented:
That would be better (returning a bool). Catching Exception is generally not good because it will catch potentially other issues, and even catching assertionError may cause confusion if the assertError is being raised by some other check than than the one you're interested in (broadcast behavior).

grafnu commented:
Ok, updated with proper bool return. Please take a look when you get a chance!
